java/integTest/src/test/java/glide/standalone/TransactionTests.java

```diff
@@ -307,6 +308,62 @@ public void watch() {
         assertInstanceOf(RequestException.class, executionException.getCause());
     }
 
+    @Test
+    @SneakyThrows
+    public void watch_binary() {
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String key3 = "{key}-3" + UUID.randomUUID();
+        String key4 = "{key}-4" + UUID.randomUUID();
+        String foobarString = "foobar";
+        String helloString = "hello";
+        GlideString[] keys =
+                new GlideString[] {GlideString.gs(key1), GlideString.gs(key2), GlideString.gs(key3)};
+        Transaction setFoobarTransaction = new Transaction();
+        Transaction setHelloTransaction = new Transaction();
+        String[] expectedExecResponse = new String[] {OK, OK, OK};
+
+        // Returns null when a watched key is modified before it is executed in a transaction command.
+        // Transaction commands are not performed.
+        assertEquals(OK, client.watch(keys).get());
+        assertEquals(OK, client.set(key2, helloString).get());
```

Comment:
nit: use binary version of `set` and `get` - apply thoughout
```suggestion
        assertEquals(OK, client.set(gs(key2), gs(helloString)).get());
```